airbyte-cdk/python/airbyte_cdk/sources/streams/checkpoint/checkpoint_reader.py

```diff
@@ -59,18 +59,19 @@ def __init__(self, stream_state: Mapping[str, Any], stream_slices: Iterable[Opti
         self._has_slices = False
 
     def next(self) -> Optional[Mapping[str, Any]]:
-        try:
-            next_slice = next(self._stream_slices)
-            self._has_slices = True
-            return next_slice
-        except StopIteration:
-            # This is used to avoid sending a duplicate state message at the end of a sync since the stream has already
-            # emitted state at the end of each slice. If we want to avoid this extra complexity, we can also just accept
-            # that every sync emits a final duplicate state
-            if self._has_slices:
-                self._state = None
+        if self._has_slices:
+            # Directly try to get the next slice without handling StopIteration here
```

Comment:
the original comments seem more useful to me because they provide context on why the StopIteration exception can be expected. The change looks fine though